hal-x86_64/src/framebuffer.rs

```diff
@@ -0,0 +1,101 @@
+use core::ptr;
+use hal_core::framebuffer::{Draw, RgbColor};
+
+#[derive(Debug)]
+pub struct Framebuffer<'buf, B> {
+    buf: B,
+    cfg: &'buf Config,
+}
```

Comment:
This might be doable in `hal-core`, but I suppose it doesn't really matter either way.

Reply:
i also considered that, and may move it in the future